client/subscriptions-empty-state/index.js

```diff
@@ -13,76 +13,87 @@ import unconnectedImage from '../../assets/images/subscriptions-empty-state-unco
-							'dashboard — powered by WooCommerce Payments.',
-						'woocommerce-payments'
-				  ) }
-		</p>
-	);
-};
+const Image = () => (
+	<img src={ isConnected ? connectedImage : unconnectedImage } alt="" />
+);
 
-const renderTOS = () => {
-	return (
-		<p className="wcpay-empty-subscriptions__tos">
-			{ createInterpolateElement(
-				__(
-					'By clicking "Finish setup", you agree to the <a>Terms of Service</a>',
+const Description = () => (
+	<p className="wcpay-empty-subscriptions__description">
+		{ isConnected
+			? __(
+					'This is where you’ll see and manage all subscriptions in your store. Create a ' +
+						'subscription product to turn one-time purchases into a steady income.',
 					'woocommerce-payments'
+			  )
+			: __(
+					'Track recurring revenue and manage active subscriptions directly from your store’s ' +
+						'dashboard — powered by WooCommerce Payments.',
+					'woocommerce-payments'
+			  ) }
+	</p>
+);
+
+const TOS = () => (
+	<p className="wcpay-empty-subscriptions__tos">
+		{ createInterpolateElement(
+			__(
+				'By clicking "Finish setup", you agree to the <a>Terms of Service</a>',
+				'woocommerce-payments'
+			),
+			{
+				a: (
+					// eslint-disable-next-line jsx-a11y/anchor-has-content
+					<a
+						href="https://wordpress.com/tos/"
+						target="_blank"
+						rel="noreferrer"
+					/>
 				),
-				{
-					a: (
-						// eslint-disable-next-line jsx-a11y/anchor-has-content
-						<a
-							href="https://wordpress.com/tos/"
-							target="_blank"
-							rel="noreferrer"
-						/>
-					),
-				}
-			) }
-		</p>
-	);
-};
+			}
+		) }
+	</p>
+);
+
+const ActionButtons = () => {
+	const [ isFinishingSetup, setIsFinishingSetup ] = useState( false );
+	const [ isCreatingProduct, setIsCreatingProduct ] = useState( false );
 
-const renderButtons = ( isConnected ) => {
 	return (
 		<div className="wcpay-empty-subscriptions__button_container">
 			{ ! isConnected && (
-				<Button isPrimary href={ wcpay.connectUrl }>
+				<Button
+					disabled={ isFinishingSetup }
+					href={ connectUrl }
+					isBusy={ isFinishingSetup }
+					isPrimary
+					onClick={ () => setIsFinishingSetup( true ) }
+				>
 					{ __( 'Finish setup', 'woocommerce-payments' ) }
 				</Button>
 			) }
-			<Button isSecondary href={ wcpay.newProductUrl }>
+			<Button
+				disabled={ isCreatingProduct }
+				href={ newProductUrl }
+				isBusy={ isCreatingProduct }
+				isSecondary
+				onClick={ () => setIsCreatingProduct( true ) }
+			>
 				{ __( 'Create subscription product', 'woocommerce-payments' ) }
 			</Button>
 		</div>
 	);
 };
 
-ReactDOM.render(
+render(
 	<div className="wcpay-empty-subscriptions__container">
-		{ renderImage( wcpay.isConnected ) }
-		{ renderDescription( wcpay.isConnected ) }
-		{ ! wcpay.isConnected && renderTOS() }
-		{ renderButtons( wcpay.isConnected ) }
+		<Image />
+		<Description />
+		{ ! isConnected && <TOS /> }
+		<ActionButtons />
```

Comment:
Rather than calling the components like a traditional JS function you almost always want to use the JSX syntax instead. i.e. `<ActionButtons />` instead of `ActionButtons()`.

More info here: https://dev.to/igor_bykov/react-calling-functional-components-as-functions-1d3l